homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -159,10 +172,46 @@ async def async_step_reauth(
         self, entry_data: Mapping[str, Any]
     ) -> ConfigFlowResult:
         """Handle configuration by re-auth."""
-        self._reauth_entry = self._get_reauth_entry()
-        if unique_id := self._reauth_entry.unique_id:
-            await self.async_set_unique_id(unique_id, raise_on_progress=False)
-        return await self.async_step_user()
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that informs the user that reauth is required."""
+        reauth_entry = self._get_reauth_entry()
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+        host = reauth_entry.data[CONF_HOST]
+
+        if user_input is not None:
+            username: str = user_input[CONF_USERNAME]
+            password: str = user_input[CONF_PASSWORD]
+            await validate_input(
+                self.hass,
+                host,
+                username,
+                password,
+                errors,
+                description_placeholders,
+            )
```

Comment:
Single use variables
```suggestion
        if user_input is not None:
            await validate_input(
                self.hass,
                reauth_entry.data[CONF_HOST],
                user_input[CONF_USERNAME],
                user_input[CONF_PASSWORD],
                errors,
                description_placeholders,
            )
```